docs/storage/s3/README.md

```diff
@@ -0,0 +1,150 @@
+# Storage
+
+Description of using static object storage as database for Application Registry.
+
+## Storing Application labels
+
+There were an idea to use Apache Parquet for storing the pieces of information regarding labels.
+We want to have an easy access to the pieces of information:
+    - list all labels for application 
+    - query application by label
+    - list all labels
+
+### S3-based Solution
+
+To achieve querying by groups, the following bucket structure is suggested:
+
+![Bucket structure](./assets/bucket-structure.jpg)
+
+Data redundancy is needed to reduce requests while doing read operations.
+
+**Runtime creation**
+
+- Create `data.json` file in `{tenant_id}/runtimes/{runtime_id}` directory (1 request)
+
+For every label of the runtime (N is the number of the runtimes):
+- Create `{key}={value}` file for every label in `{tenant_id}/runtimes/{runtime_id}/labels` (N requests)
+- Create `data.json` file in `{tenant_id}/labels/{key}/{value}/runtimes/{runtime_id}` directory (N requests)
+- Create `key=value` file in `{tenant_id}/labels/{key}/{value}/runtimes/{runtime_id}/labels` directory (N * N requests)
+
+Total calls: N^2 + 2*N + 1
+
+**Application registration**
+
+Similar to Runtime creation - replace `runtimes` with `applications`
+
+Total calls: N^2 + 2*N + 1
+
+**Adding label to Application**
+
+- Create file `{key}={value}` in `{tenant_id}/applications/{application_id}` directory (1 request)
```

Comment:
Why adding label to app has less calls than doing same with runtime?

Reply:
I've covered runtime creation, and this is the case of adding label to an existing application. I will add another sections to cover all cases for app and runtime